src/criteria/default/index.php

```diff
@@ -0,0 +1,121 @@
+<?php
+/**
+ * Register default segmentation criteria.
+ *
+ * @package Newspack
+ */
+
+namespace Newspack_Popups;
+
+defined( 'ABSPATH' ) || exit;
+
+$criteria = [
```

Comment:
Nit: in the future, when we want to consider using this system register additional criteria from other plugins of ours, wouldn't we want to apply a filter to this `$criteria` object so other plugins could add their own criteria by simply appending new keys and config to the object, and/or remove default criteria by unsetting default keys? I see the `newspack_popups_registered_criteria` filter [here](https://github.com/Automattic/newspack-popups/pull/1155/files#diff-8da51b859e12deff15c7b71debad6c7e6859326edc7d11c8c56630d57215b816R98), but being able to hook into the config before the criteria are registered seems more useful to me.